Fail to buid against GHCJS(or maybe other 32bit platform)

Build fail with ghcjs-8.6.0.1, due to type mismatch. `ltWord#` & `eqWord#` accept `Word#` type, but `Word#` type is not always `Word64#`.

Below is part of of build message:

```
[1 of 3] Compiling Data.TypeRepMap.Internal ( src/Data/TypeRepMap/Internal.hs, dist/build/Data/TypeRepMap/Internal.js_o )

src/Data/TypeRepMap/Internal.hs:328:69: error:
    • Couldn't match expected type ‘GHC.Prim.Word#’
                  with actual type ‘GHC.Prim.Word64#’
    • In the first argument of ‘ltWord#’, namely ‘a’
      In the expression: a `ltWord#` valA
      In the expression:
        case a `ltWord#` valA of
          0#
            -> case a `eqWord#` valA of
                 0# -> go (2# *# i +# 2#)
                 _ -> let ... in ...
          _ -> go (2# *# i +# 1#)
    |

```

--- a/src/Data/TypeRepMap/Internal.hs
+++ b/src/Data/TypeRepMap/Internal.hs
@@ -25,6 +25,8 @@ If you need to use this module for purposes other than tests,
 create an issue.
 -}
 
+#include "MachDeps.h"
+
 module Data.TypeRepMap.Internal where
 
 import Prelude hiding (lookup)
@@ -44,7 +46,13 @@ import Data.Semigroup (Semigroup (..), All(..))
 import GHC.Base (Any, Int (..), Int#, (*#), (+#), (<#))
 import GHC.Exts (IsList (..), inline, sortWith)
 import GHC.Fingerprint (Fingerprint (..))
+#if WORD_SIZE_IN_BITS >= 64
 import GHC.Prim (eqWord#, ltWord#)
+#else
+import GHC.IntWord64 (eqWord64#, ltWord64#)
+#define eqWord eqWord64
+#define ltWord ltWord64
+#endif
 import GHC.Word (Word64 (..))
 import Type.Reflection (SomeTypeRep (..), TypeRep, Typeable, typeRep, withTypeable)
 import Type.Reflection.Unsafe (typeRepFingerprint)

